test/e2e/framework/util.go

```diff
@@ -3055,6 +3055,22 @@ func WaitForPodsReady(c clientset.Interface, ns, name string, minReadySeconds in
 	})
 }
 
+// Waits for an event on the given object generated for the desired reason and containing the given message
+func WaitForParticularEvent(c clientset.Interface, ns string, objOrRef runtime.Object, desiredReason, substrMessage string) error {
+	return wait.Poll(Poll, 5*time.Minute, func() (bool, error) {
+		events, err := c.Core().Events(ns).Search(api.Scheme, objOrRef)
+		if err != nil {
+			return false, fmt.Errorf("error in listing events: %s", err)
+		}
+		for _, e := range events.Items {
+			if e.Reason == desiredReason && strings.Contains(e.Message, substrMessage) {
```

Comment:
do you want to massage `desiredReason`, eg. trim spaces, down-case, etc? If the desired reasons are defined by string constants then I'd say no, but if the caller uses their own strings then maybe.